ocaml/forkexecd/src/child.ml

```diff
@@ -59,28 +59,42 @@ let handle_comms_sock comms_sock state =
       state
 
 let handle_comms_no_fd_sock2 comms_sock fd_sock state =
-  debug "Selecting in handle_comms_no_fd_sock2" ;
-  let ready, _, _ = Unix.select [comms_sock; fd_sock] [] [] (-1.0) in
-  debug "Done" ;
-  if List.mem fd_sock ready then (
-    debug "fd sock" ;
-    let fd_sock2, _ = Unix.accept fd_sock in
-    {state with fd_sock2= Some fd_sock2}
-  ) else (
-    debug "comms sock" ;
-    handle_comms_sock comms_sock state
+  debug "Using epoll in handle_comms_no_fd_sock2" ;
+  let epoll = Polly.create () in
+  List.iter
+    (fun fd -> Polly.add epoll fd Polly.Events.inp)
+    [comms_sock; fd_sock] ;
+  (* Although there are two fds, we set max_fds to 1 here as we only want this
+     function to trigger once so that we get one return value *)
+  Polly.wait_fold epoll 1 (-1) state (fun _ fd _ _ ->
```

Comment:
Is 1 correct here though, would that prevent it from even watching the other FD?

Reply:
My understanding is that one FD would be processed per round - and any other FD that is ready would be reported again in the next round. But does this guarantee fairness?

Reply:
This was the only way I could think to make it work, do you have any other alternatives?